android/library/src/main/java/io/logto/android/authflow/webview/AuthWebViewClient.kt

```diff
@@ -0,0 +1,43 @@
+package io.logto.android.authflow.webview
+
+import android.app.Activity
+import android.util.Log
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import io.logto.android.constant.AuthConstant
+import io.logto.android.utils.UrlUtil
+
+class AuthWebViewClient(
+    private val attachedActivity: Activity,
+    private val specifiedRedirectUri: String
+) : WebViewClient() {
+
+    override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+        val redirectUrl = request!!.url.toString()
```

Comment:
don't use `!!` since it's unsafe, would you mind to add a lint rule for it?

Reply:
`!!` will thorw a `NullPointerException` if `request` is null.
this is like:
```kotlin
var redirectUrl: String? = null
if (request != null) {
  redirectUrl = request.url.toString()
} elset {
  throw NullPointerException
}
```
need more research, issue is ready:
https://linear.app/silverhand/issue/LOG-78/add-lint-rule-to-avoid-usage

Reply:
i know, thus we need either explicitly throw our customized error to specify the reason or to accept the optional result, instead of a less meaningful `NullPointerException`.

Reply:
OK